holidays/countries/cocos_islands.py

```diff
@@ -0,0 +1,237 @@
+        * [2024](https://web.archive.org/web/20250207203100/https://www.infrastructure.gov.au/sites/default/files/documents/a12-2023-2024-public-holidays-cocos-k-islands.pdf)
+        * [2025](https://web.archive.org/web/20250413083314/https://www.infrastructure.gov.au/sites/default/files/documents/a21-2024-administrator-community-bulletin-cki-public-holidays-2025.pdf)
+    """
+
+    country = "CC"
+    default_language = "en_CC"
+    # %s (observed).
+    observed_label = tr("%s (observed)")
+    supported_languages = ("coa_CC", "en_CC", "en_US")
+    # Act of Self Determination 1984.
+    start_year = 1985
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, CocosIslandsStaticHolidays)
+        IslamicHolidays.__init__(
+            self, cls=CocosIslandsIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("New Year's Day")))
+
+        # Australia Day.
+        self._add_observed(self._add_holiday_jan_26(tr("Australia Day")))
+
+        # Act of Self Determination Day.
+        self._add_observed(self._add_holiday_apr_6(tr("Act of Self Determination Day")))
+
+        # Good Friday.
+        self._add_observed(self._add_good_friday(tr("Good Friday")))
+
+        # Easter Monday.
+        self._add_observed(self._add_easter_monday(tr("Easter Monday")))
+
+        # ANZAC Day.
+        self._add_observed(self._add_holiday_apr_25(tr("ANZAC Day")))
+
+        self._add_holiday_2nd_mon_of_jun(
+            # King's Birthday.
+            tr("King's Birthday")
+            if self._year >= 2023
+            # Queen's Birthday.
+            else tr("Queen's Birthday")
+        )
+
+        # Placed before Christmas Day for proper observed calculation.
+        self._add_observed(
+            # Boxing Day.
+            self._add_christmas_day_two(tr("Boxing Day")),
+            rule=SAT_SUN_TO_NEXT_MON_TUE + MON_TO_NEXT_TUE,
+        )
+
+        # Christmas Day.
+        self._add_observed(self._add_christmas_day(tr("Christmas Day")))
+
+        # Islamic holidays.
+
+        if self._year <= 2019:
+            # Islamic New Year.
+            for dt in self._add_islamic_new_year_day(tr("Islamic New Year")):
+                self._add_observed(dt, tr("Islamic New Year"))
```

Comment:
```suggestion
                self._add_observed(dt)
```
Look at other countries examples.